lib/elixir/lib/kernel/lexical_tracker.ex

```diff
@@ -171,8 +172,21 @@ defmodule Kernel.LexicalTracker do
   defp add_compile(d, module, :runtime), do: :ets.insert_new(d, {{:mode, module}, :runtime})
   defp add_compile(d, module, :compile), do: :ets.insert(d, {{:mode, module}, :compile})
 
-  defp add_directive(d, module, line, warn, tag) do
+  defp add_directive(d, m_or_mfa, line, warn, :import = tag) when is_atom(m_or_mfa) do
```

Comment:
I'm not happy with this impl. but couldn't figure out how to improve it yet. This is needed for this test case: https://github.com/wojtekmach/elixir/blob/18cf5cdb5363000472c2e5805fc583f5e39a692a/lib/elixir/test/elixir/kernel/lexical_tracker_test.exs#L63:L68